mod/groups/lib/groups.php

```diff
@@ -51,12 +51,10 @@ function groups_prepare_form_vars($group = null) {
 	}
 
 	// handle tool options
-	$tools = elgg_get_config('group_tool_options');
-	if ($tools) {
-		foreach ($tools as $group_option) {
-			$option_name = $group_option->name . "_enable";
-			$values[$option_name] = $group_option->default_on ? 'yes' : 'no';
-		}
+	$tools = groups_get_group_tool_options($group);
```

Comment:
As this will be type-hinted, make sure $group is `null` if it isn't ElggGroup. `false` will fail.